service/controller/resource/instance/create_cordon_old_workers.go

```diff
@@ -131,22 +134,31 @@ func sortNodesByTenantVMState(nodes []corev1.Node, instances []compute.VirtualMa
 		nodeMap[n.GetName()] = n
 	}
 
+	myVersion := semver.New(project.Version())
+
 	for _, i := range instances {
 		name := instanceNameFunc(customObject, *i.InstanceID)
 
-		if *i.LatestModelApplied {
-			n, found := nodeMap[name]
+		n, found := nodeMap[name]
+		if !found {
 			// When VMSS is scaling up there might be VM instances that haven't
 			// registered as nodes in k8s yet. Hence not all instances are
 			// found from node list.
-			if found {
-				newNodes = append(newNodes, n)
-			}
+			continue
+		}
+
+		v, exists := n.GetLabels()[label.OperatorVersion]
+		if !exists {
+			// Label does not exist, we consider node as old.
+			oldNodes = append(oldNodes, n)
+			continue
+		}
+
+		nodeVersion := semver.New(v)
+		if nodeVersion.LessThan(*myVersion) {
```

Comment:
I've been thinking about this. I think I used `LessThan()` as well, but I wonder if just difference (`nodeVersion != myVersion`) would be enough or better? Our code in general doesn't care that much about semantics of versions, but rather that all nodes would be same version.

Reply:
That would make sense if we supported downgrading of nodes. But we don't so I guess there's no difference in using `!=` or `LessThan`, right?

Reply:
Yes, both work so no difference. All good.